snmp/datadog_checks/snmp/snmp.py

```diff
@@ -196,11 +197,20 @@ def check_table(self, config, table_oids):
                 # No table to browse, just one symbol
                 oids.append(table)
             elif len(symbols) < bulk_limit:
-                oids.extend(symbols)
+                all_oids.append(symbols)
             else:
                 bulk_oids.append(table)
 
-        all_binds, error = self.fetch_oids(config, oids, enforce_constraints=enforce_constraints)
```

Comment:
> Trying to use a single next command to fetch all table oids can lead to
misleading results.

What are those "misleading results" ?

Reply:
OK, so it's a little bit complicated. Imagine you're browsing table A with 10 elements, and table B with 12. If you pass both in in the same next command, when you reach elements 11 and 12, pysnmp < 4.4.5 is going to give you the elements which are next to table A in the tree (could be anything). Those are going to be in the results, but we're going to filter them out because the metrics don't exist.

In pysnmp >= 4.4.5, we actually get an error, because it's filtering out those values that are next.

So I think the best fix is not to query them, and fetch the OIDs per table. There is no noticeable performance change that I could see. If I disable bulk, it actually seems to be a bit faster  because we query less values.

Reply:
Thx for the explanation.

1. So, it means it's not safe to mix elements from 2 tables using next command?

2. With current fix seems to fetch oids one by one (https://github.com/DataDog/integrations-core/pull/5192/files#diff-c232c5e7cec47d95c2ccdffda2acf33bR210). Isn't this an issue performance wise ?

I guess, because of 1. We don't have the choice, right? or fetch the OIDs per table as you suggested.

3. If that's possible, could you add a test for this case?

Reply:
1. That's my understanding yes.
2. No we fetch tables per tables. We still pass a list. It doesn't seem to have a big impact on the performance (you can run the benchmarks locally to get some ideas).
3. That's super tricky. This fix is validated by upgrade pysnmp.